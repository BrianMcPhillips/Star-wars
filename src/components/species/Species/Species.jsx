import React from 'react';
import PropTypes from 'prop-types';

const Species = ({ name, url }) => (
  <>
    <h1>{name}</h1>
  </>
);

Species.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Species;
