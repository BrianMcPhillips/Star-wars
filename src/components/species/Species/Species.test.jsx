import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Species from './Species';

describe('Species component', () => {
  afterEach(() => cleanup());
  it('renders Species', () => {
    const { asFragment } = render(
      <Species 
        name="Human"
        url="https://swapi.dev/api/species/1/"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
