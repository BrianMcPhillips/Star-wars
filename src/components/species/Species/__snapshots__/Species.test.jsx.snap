// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Species component renders Species 1`] = `
<DocumentFragment>
  <h1>
    Human
  </h1>
</DocumentFragment>
`;
